packages/@aws-cdk-testing/cli-integ/tests/cli-integ-tests/cli.integtest.ts

```diff
@@ -1318,6 +1319,45 @@ integTest(
   }),
 );
 
+integTest('deploy stack with Lambda Asset to Object Lock-enabled asset bucket', withoutBootstrap(async (fixture) => {
+  // Bootstrapping with custom toolkit stack name and qualifier
+  const qualifier = fixture.qualifier;
+  const toolkitStackName = fixture.bootstrapStackName;
+  await fixture.cdkBootstrapModern({
+    verbose: true,
+    toolkitStackName: toolkitStackName,
+    qualifier: qualifier,
+  });
+
+  try {
+    const bucketName = `cdk-${qualifier}-assets-${await fixture.aws.account()}-${fixture.aws.region}`;
+    await fixture.aws.s3.send(new PutObjectLockConfigurationCommand({
+      Bucket: bucketName,
+      ObjectLockConfiguration: {
+        ObjectLockEnabled: 'Enabled',
+        Rule: {
+          DefaultRetention: {
+            Days: 1,
+            Mode: 'GOVERNANCE',
+          },
+        },
+      },
+    }));
+
+    // Deploy a stack that definitely contains a file asset
+    await fixture.cdkDeploy('lambda', {
+      options: [
+        '--toolkit-stack-name', toolkitStackName,
+        '--context', `@aws-cdk/core:bootstrapQualifier=${qualifier}`,
+      ],
+    });
+
+    // THEN - should not fail
+  } finally {
+    // Clean up stack and bootstrap stack
```

Comment:
No cleanup?

Reply:
Happens automatically. I should remove the `finally`.